packages/opentelemetry-core/src/utils/error.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function ensureError(unknownError: unknown): Error {
+    let error: Error;
+    if (unknownError instanceof Error) {
```

Comment:
The `instanceof Error` check can be false if cross-context execution like iframe or Node.js `vm` is involved. I think here it could be better to check on shape, like:

```typescript
function isErrorLike(value: unknown): value is Error {
  const it = value as any;
  return it != null && typeof it.message === 'string' && (it.stack == null || typeof it.stack === 'string');
}
```